exporter/fileexporter/config.go

```diff
@@ -79,3 +85,34 @@ func (cfg *Config) Validate() error {
 	}
 	return nil
 }
+
+// Unmarshal a confmap.Conf into the config struct.
+func (cfg *Config) Unmarshal(componentParser *confmap.Conf) error {
+	if componentParser == nil {
+		return errors.New("empty config for file exporter")
+	}
+	// first load the config normally
+	err := componentParser.UnmarshalExact(cfg)
+	if err != nil {
+		return err
+	}
+
+	// next manually search for protocols in the confmap.Conf,
+	// if rotation is not present it means it is disabled.
+	if !componentParser.IsSet(rotationFieldName) {
+		return nil
+	}
+	rotation := componentParser.Get(rotationFieldName)
+	if rotation == nil {
```

Comment:
When would this case happen? We just checked `IsSet(rotationFieldName)`, so maybe this can't happen?

Reply:
If the user specifies rotation and no other configuration, `cfg.Rotation `is nil, `rotationCfg.IsSet(backupsFieldName) ` returns false, `cfg.Rotation.MaxBackups = defaultMaxBackups` will cause panic.